src/core_plugins/kibana/public/management/sections/indices/_create.js

```diff
@@ -247,6 +254,18 @@ uiModules.get('apps/management')
     }, notify.fatal);
   }
 
+  function updateFieldListAndSetTimeField(results, seedField) {
+    updateFieldList(results);
+
+    if (!results.dateFields.length) {
+      return;
+    }
+
+    const matchingTimeField = results.dateFields.find(field => field.name === seedField);
+    const defaultTimeField = results.dateFields[0];
+    index.timeField = matchingTimeField ? matchingTimeField : defaultTimeField;
```

Comment:
Maybe I'm missing something... is index.timeField mutated? If so, then it makes sense to assign a "fresh" reference.